src/libraries/Microsoft.PowerFx.Connectors/TabularExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.UtilityDataStructures;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public static class TabularExtensions
+    {
+        public static RecordType AddAssociatedDataSource(this RecordType recordType, DName name, string datasetName, ServiceCapabilities serviceCapabilities, bool isReadOnly, IDictionary<string, string> displayNameMapping = null)
```

Comment:
>AddAssociatedDataSource [](http://example.com/codeflow?start=33&length=23)

This one is called by ServiceNow code

Reply:
can this be removed now? 
since SN will pass capability information through the RecordType ctor.

Reply:
SN doesn't use this API anymore
Anyhow, AddTabularDataSource still uses it